knack/invocation.py

```diff
@@ -139,6 +145,28 @@ def execute(self, args):
 
         params = self._filter_params(parsed_args)
 
+        cmd = parsed_args.func
+        deprecations = [] + getattr(parsed_args, '_argument_deprecations', [])
+        if cmd.deprecate_info:
+            deprecations.append(cmd.deprecate_info)
+
+        # search for implicit deprecation
+        path_comps = cmd.name.split()[:-1]
+        implicit_deprecate_info = None
+        while path_comps and not implicit_deprecate_info:
+            implicit_deprecate_info = resolve_deprecate_info(self.cli_ctx, ' '.join(path_comps))
+            del path_comps[-1]
+
+        if implicit_deprecate_info:
+            deprecate_kwargs = implicit_deprecate_info.__dict__.copy()
+            deprecate_kwargs['object_type'] = 'command'
+            del deprecate_kwargs['_get_tag']
+            del deprecate_kwargs['_get_message']
+            deprecations.append(ImplicitDeprecated(**deprecate_kwargs))
+
+        for d in deprecations:
+            logger.warning(str(d.message))
```

Comment:
Is the `str()` necessary

Reply:
Yes.

Reply:
Doesn't the logger.warning method call the string method of whatever is passed?
If it doesn't it probably should..

Reply:
It doesn't and it's not under our control--hence the call to `str(...)`